pkg/labels/v2/labels.go

```diff
@@ -174,6 +177,44 @@ func (lbls Labels) String() string {
 // optimizing for smallest memory use.
 const smallLabelsSize = 9
 
+func (lbls Labels) MarshalJSON() ([]byte, error) {
```

Comment:
Random thought: i used `github.com/json-iterator/go` elsewhere and it was very fast with very little effort. Maybe worth a quick explore?